capsules/src/console.rs

```diff
@@ -204,15 +205,18 @@ impl<'a, U: UART> Client for Console<'a, U> {
         // Either print more from the AppSlice or send a callback to the
         // application.
         self.tx_buffer.replace(buffer);
-        self.in_progress.take().map(|appid| {
+        self.in_progress.get().map(|appid| {
+            self.in_progress.set(None);
```

Comment:
Agreed on both fronts, but neither are sufficient to stand in the way of merging this as-is right now IMO. I think we should pull the trigger.